### ISSUE TITLE
Client has recently stopped working with "Failed to get Authorization code" (Failed to create rest session)

### ISSUE THREAD

**Issue by davidbairdala** [bullhorn/sdk-rest#397]
I am using the Bullhorn Java SDK to retrieve candidate data. Up until this morning I was able to authenticate and execute filter searches by creating a new StandardBullhornData instance. 

Now the creation of the StandardBullHornData class fails with an "Unable to create rest client" exception. Looking at the stack trace I can see that the underlying cause is a  null pointer exception in RestApiSession.getAuthCode() because a URI being returned from restTemplate.postForLocation() is null. It appears to me that this method (postForLocation) is always expecting a 302 response with a "Location" header, but when I examine the response for the request (the third request in the authentication sequence) the status code is 200, and there is no Location header, resulting in a null pointer exception

It is entirely possible that an infrastructure issue is causing this, but I have no way of telling what is going wrong.

I am using the following URLs:

REST token URL: https://rest-aus.bullhornstaffing.com/oauth/token
REST Authorize URL: https://rest-aus.bullhornstaffing.com/oauth/authorize
REST Login URL: https://rest-aus.bullhornstaffing.com/rest-services/login

I have tested with versions 1.5.47 and 1.4.49 of the SDK, and I have not changed the credentials, client secret or anything else.

It seems to me that at the very least that the SDK should deal with the 200 response

**Comment by davidbairdala** [comment]
Some additional information - I can authenticate manually using Postman, so I am fairly confident that there is nothing wrong with the credentials I am using. Some assistance would be greatly appreciated.

**Comment by pbearne** [comment]
Are you getting bad rest token or time-out message?

I have seen this as well but found the solution.

**Comment by davidbairdala** [comment]
No, no timeout message or anything about a bad rest token. Just the message "Failed to get Authorization code", and a stacktrace that indicates that a Null Pointer Exception is the root cause.

**Comment by davidbairdala** [comment]
Some more additional information. Like I said before, if I use postman to GET the authorize URL I get a 200-OK, and no Location header. If I hit the authorize URL using chrome, however, the server returns 302, and the browser follows the redirect to a bullhorn.com URL with an authorization code as a parameter (it actually follows 3 redirects before it finally gets a 200 response).

If I use "get" or "curl" to do the GET I get the following error(s) (I haven't posted actual URLs because they contain secrets):

wget
`
Resolving rest-aus.bullhornstaffing.com (rest-aus.bullhornstaffing.com)... 3.105.56.101
Connecting to rest-aus.bullhornstaffing.com (rest-aus.bullhornstaffing.com)|3.105.56.101|:443... connected.
OpenSSL: error:0A000152:SSL routines::unsafe legacy renegotiation disabled
Unable to establish SSL connection.
`

curl (verbose)
`
*   Trying 3.105.56.101:443...
* Connected to rest-aus.bullhornstaffing.com (3.105.56.101) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (OUT), TLS header, Unknown (21):
* TLSv1.2 (OUT), TLS alert, handshake failure (552):
* error:0A000152:SSL routines::unsafe legacy renegotiation disabled
* Closing connection 0
curl: (35) error:0A000152:SSL routines::unsafe legacy renegotiation disabled
`
These commands were performed on a fully patched PopOS 22-04 LTS. I think these errors are telling me that the OpenSSL implementation on the client is rejecting the SSL handshake from the server due to a legacy (unsafe) protocol.

**Comment by davidbairdala** [comment]
So I tried something different. I tried hitting the authorize URL in Chrome, but in a new incognito tab. Instead of redirecting to a URL with a code in it, it presented a challenge screen asking to accept or deny the request. I clicked accept, and since then my tests now pass. So it seems the 200 that the client was getting was actually this challenge screen.

**Comment by pbearne** [comment]
I have seen that problem 
it still catches me as well

Paul
